### PR TITLE
Deprecate OptionSetInfo, prepare way for new ChoiceInfo function

### PR DESCRIPTION
**Pull request by gregli-msft**
OptionSetInfo only returns one value, the logical name of an option set choice.  We'd like to redesign this function to accommodate:

1. The logical name is a text wrapped number for number and Boolean backed option sets coming from Dataverse.  We recently added the ability to get at these values with the Value and Boolean functions which is a better way, strongly typed.  https://github.com/microsoft/Power-Fx/pull/2230
1. There are other pieces of information that are worth knowing, for example the color of an option set choice, that can't be accommodated with this design.  
1. Our preferred design is to return an extensible record of information.  We could add an additional second argument with an enum to determine what information to return, with a default value of logical name, but this isn't our first choice.
1. We don't use the term OptionSet any longer in Dataverse, it has been replaced with Choice, which is more consistent with our existing Choices function.

As some hosts may still be using this function, it will remain available as a function that can be enabled by calling config.EnableOptionSetInfo.  Value/Boolean can be used immediately as a workaround, wrapped with Text if needed.  We plan to add a new ChoiceInfo function with a separate PR.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Environment/PowerFxConfigExtensions.cs [hunk 1]
```diff
@@ -60,5 +61,11 @@ public static void EnableRegExFunctions(this PowerFxConfig config, TimeSpan regE
                 config.AdditionalFunctions.Add(func.Key, func.Value);
             }
         }
+
+        [Obsolete("OptionSetInfo function is deprecated. Use the Value function on an option set backed by a number and the Boolean function on an option set bacekd by a Boolean instead.  A new ChoiceInfo function is in the works for access to logical names.")]
```
Comment (LucGenetier) [inline src/libraries/Microsoft.PowerFx.Interpreter/Environment/PowerFxConfigExtensions.cs:65]: nit: typo on bacekd

**Review by LucGenetier — APPROVED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/OptionSetInfo.cs [hunk 2]
```diff
@@ -2,13 +2,18 @@
...
 using Microsoft.PowerFx.Core.Localization;
 using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
 
-namespace Microsoft.PowerFx.Core.Texl.Builtins
+namespace Microsoft.PowerFx.Functions
 {
-    internal sealed class OptionSetInfoFunction : BuiltinFunction
+    internal sealed class OptionSetInfoFunction : BuiltinFunction, IAsyncTexlFunction
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/OptionSetInfo.cs:16]: We should not have any function implementation inside Microsoft.PowerFx.Core. This should be moved to the MS.PFx.Interpreter instead

Comment (gregli-msft) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/OptionSetInfo.cs]: Done.

File: src/libraries/Microsoft.PowerFx.Interpreter/Environment/PowerFxConfigExtensions.cs (hunk 1 above)
Comment (gregli-msft) [reply src/libraries/Microsoft.PowerFx.Interpreter/Environment/PowerFxConfigExtensions.cs]: Done.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/OptionSetInfo.cs [hunk 3]
```diff
@@ -23,5 +28,20 @@ public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
         {
             yield return new[] { TexlStrings.AboutOptionSetInfoArg1 };
         }
+
+        public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (gregli-msft) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/OptionSetInfo.cs]: Done.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Pull request merged**